Container docs rewrite and reorganization (#14527)
microsoft/FluidFramework: docs/content/docs/build/container-states-events/index.md

But the new container has a different ID from the deleted one and subsequent clients will have to connect to it using the new ID.

#### Handling publication status

Your code can test for the publication status with the [container.AttachState]({{< relref "/docs/apis/fluid-static/fluidcontainer-class.md#attachstate-property" >}}) property which has an [AttachState]({{< relref "/docs/apis/container-definitions.md#attachstate-enum" >}}) value. 
This can be useful if your application will publish the container in some code paths on the creating client, but not others. 
For example, on the creating computer, you don't want to call `container.attach` if it has already been called. In simple cases, you can know at coding time if that has happened, but when the creating client in complex code flows and calls of `container.attach` appear in more than one branch, you may need to test for this possibility. The following is a simple example.

```typescript
// Code that runs only on a creating client.
if (container.attachState !== AttachState.Attached) {
    await container.attach();
}
```

> How you handle the **publishing** (`AttachState.Attaching`) state depends on the situation. This state implies that `container.attach` has already been called, so to prevent a recall treat publishing (`Attaching`) the same as published (`Attached`). The following is an example.

Reviewer: ```suggestion
How you handle the **publishing** (`AttachState.Attaching`) state depends on the situation. This state implies that `container.attach` has already been called, so to prevent an additional call treat publishing (`Attaching`) the same as published (`Attached`). The following is an example.
```
Clearer wording, I think?

```typescript

> // Code that runs only on a creating client.
> if ((container.attachState !== AttachState.Attached) 
>     || 
>     (container.attachState !== AttachState.Attaching)) {
>     await container.attach();
> }

Reviewer: ```suggestion
// Code that runs only on a creating client.
if ((container.attachState !== AttachState.Attached) 
    &&
    (container.attachState !== AttachState.Attaching)) {
    await container.attach();
}
```

```

> On the other hand, in scenarios where you want to block users from editing shared data when the container is unpublished, you also want to block them when the container is in **publishing** (`AttachState.Attaching`) state, since at that point the container is not fully published. So in those scenario, treat publishing (`Attaching`) the same as unpublished (`Detached`). The following is an example.

Reviewer: ```suggestion
On the other hand, in scenarios where you want to block users from editing shared data when the container is unpublished, you also want to block them when the container is in **publishing** (`AttachState.Attaching`) state, since at that point the container is not fully published. So in those scenarios, treat publishing (`Attaching`) the same as unpublished (`Detached`). The following is an example.
```
